map/apply-scheduled-updates-to-preplanned-map-area/src/main/java/com/esri/samples/apply_scheduled_updates_to_preplanned_map_area/ApplyScheduledUpdatesToPreplannedMapAreaSample.java

```diff
@@ -0,0 +1,274 @@
+                applyUpdatesButton.setOnAction(e -> {
+
+                  // show progress indicator
+                  progressIndicator.setVisible(true);
+
+                  // create default parameters for the sync task
+                  ListenableFuture<OfflineMapSyncParameters> offlineMapSyncParametersFuture = offlineMapSyncTask.createDefaultOfflineMapSyncParametersAsync();
+                  offlineMapSyncParametersFuture.addDoneListener(() -> {
+                    try {
+                      OfflineMapSyncParameters offlineMapSyncParameters = offlineMapSyncParametersFuture.get();
+
+                      // set the sync direction to none, since we only want to update
+                      offlineMapSyncParameters.setSyncDirection(SyncGeodatabaseParameters.SyncDirection.NONE);
+                      // set the parameters to download all updates for the mobile map packages
+                      offlineMapSyncParameters.setPreplannedScheduledUpdatesOption(PreplannedScheduledUpdatesOption.DOWNLOAD_ALL_UPDATES);
+                      // set the map package to rollback to the old state should the sync job fail
+                      offlineMapSyncParameters.setRollbackOnFailure(true);
+
+                      // create a sync job using the parameters
+                      OfflineMapSyncJob offlineMapSyncJob = offlineMapSyncTask.syncOfflineMap(offlineMapSyncParameters);
+
+                      // start the job and get the results
+                      offlineMapSyncJob.start();
+                      offlineMapSyncJob.addJobDoneListener(() -> {
+                        if (offlineMapSyncJob.getStatus() == Job.Status.SUCCEEDED) {
+                          OfflineMapSyncResult offlineMapSyncResult = offlineMapSyncJob.getResult();
+
+                          // if mobile map package reopen is required, close the existing mobile map package and load it again
+                          if (offlineMapSyncResult.isMobileMapPackageReopenRequired()) {
+                            mobileMapPackage.close();
+                            mobileMapPackage.loadAsync();
+                            mobileMapPackage.addDoneLoadingListener(() -> {
+                              if (mobileMapPackage.getLoadStatus() == LoadStatus.LOADED && !mobileMapPackage.getMaps().isEmpty()) {
+
+                                // add the map from the mobile map package to the map view
+                                mapView.setMap(mobileMapPackage.getMaps().get(0));
+
+                              } else {
+                                new Alert(Alert.AlertType.ERROR, "Failed to load the mobile map package.").show();
+                              }
+                            });
+                          }
+
+                          // perform an additional check for updates
+                          ListenableFuture<OfflineMapUpdatesInfo> offlineMapUpdatesInfoFuture2 = offlineMapSyncTask.checkForUpdatesAsync();
```

Comment:
much of this code is a repeat of the above: can abstract most of this code out into a separate method. This avoids the use of a variable name with a number at the end, we should try to avoid that for readability.